src/main/java/org/cirdles/topsoil/dataset/Dataset.java

```diff
@@ -33,6 +33,8 @@
 
     public Optional<String> getName();
 
+    public void setName(String name);
```

Comment:
Or `SimpleDataset` could generate a name for itself without a `name` argument. Example: `dataset-20150617122300`